### PR TITLE
Cleanup safe core test cases

### PR DESCRIPTION
**Pull request by deepesh-kn**
This PR includes the following changes.
- Deleted old test files that had all the tests in single file and interdependent test.
- Created new test files for each function of SafeCore contract.

fixes: #535

### CODE REVIEW

File: test/gateway/safe_core/get_state_root.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+// http://www.simpletoken.org/
+//
+// ----------------------------------------------------------------------------
+
+const SafeCore = artifacts.require("./SafeCore.sol");
+const MockMembersManager = artifacts.require('MockMembersManager.sol');
+const web3 = require('../../test_lib/web3.js');
+const BN = require('bn.js');
+
+contract('SafeCore.getStateRoot()', function (accounts) {
```
Comment (gulshanvasnani) [inline test/gateway/safe_core/get_state_root.js]: Add test case to verify that it returns zero bytes for blockheight which is not stored.

File: test/gateway/safe_core/set_co_core_address.js [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    safeCore = await SafeCore.new(
+      remoteChainId,
+      blockHeight,
+      stateRoot,
+      membersManager.address,
+    );
+
+  });
+
+
```
Comment (gulshanvasnani) [inline test/gateway/safe_core/set_co_core_address.js]: Remove an extra newline.

File: test/gateway/safe_core/set_co_core_address.js [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+  it('should fail to set coCore address zero', async () => {
```
Comment (gulshanvasnani) [inline test/gateway/safe_core/set_co_core_address.js]: should fail to set coCore address zero ==> should fail when coCore address is set to zero

File: test/gateway/safe_core/commit_state_root.js [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+      ),
+      'Given block height is lower or equal to highest committed state root block height.',
+    );
+
+  });
+
+  it('should fail when block height is equal to the latest committed ' +
+    'state root\'s block height', async () => {
+
+    blockHeight = new BN(5);
```
Comment (schemar) [inline test/gateway/safe_core/commit_state_root.js]: this should not be necessary. it could even harm the test if the block height is changed in `beforeEach`

File: test/gateway/safe_core/commit_state_root.js [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+      ),
+      'State root must not be zero.',
+    );
+
+  });
+
+  it('should fail when block height is less than the latest committed ' +
+    'state root\'s block height', async () => {
+
+    blockHeight = new BN(4);
```
Comment (schemar) [inline test/gateway/safe_core/commit_state_root.js]: decreasing the block height is safer, e.g. in case where it is changed in `beforeEach`

File: test/gateway/safe_core/get_latest_state_root_block_height.js [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+      blockHeight.eq(latestBlockHeight),
+      true,
+      `Latest block height from the contract must be ${blockHeight}.`,
+    );
+
+  });
+
+  it('should return the latest committed state root block height', async () => {
+
+    blockHeight = new BN(50000);
```
Comment (schemar) [inline test/gateway/safe_core/get_latest_state_root_block_height.js]: You could `add` instead of hard-coding it.

File: test/gateway/safe_core/get_state_root.js [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+      latestStateRoot,
+      stateRoot,
+      `Latest state root from the contract must be ${stateRoot}.`,
+    );
+
+  });
+
+  it('should return the zero bytes for non committed block heights', async () => {
+
+    blockHeight = new BN(500);
```
Comment (schemar) [inline test/gateway/safe_core/get_state_root.js]: same

File: test/gateway/safe_core/get_state_root.js [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+      latestStateRoot,
+      zeroBytes,
+      `Latest state root from the contract must be ${zeroBytes}.`,
+    );
+
+  });
+
+  it('should return the latest committed state root', async () => {
+
+    blockHeight = new BN(50000);
```
Comment (schemar) [inline test/gateway/safe_core/get_state_root.js]: same

File: test/gateway/safe_core/set_co_core_address.js [hunk 9]
```diff
@@ -0,0 +1,117 @@
...
+    await Utils.expectRevert(
+      safeCore.setCoCoreAddress(coCoreAddress, { from: owner }),
+      'Co-Core address must not be 0.',
+    );
+
+  });
+
+  it('should fail when caller is not organisation owner', async () => {
+
+    owner = accounts[7];
```
Comment (schemar) [inline test/gateway/safe_core/set_co_core_address.js]: this is not the owner. maybe call it `notOwner` or something?

File: test/gateway/safe_core/set_co_core_address.js [hunk 10]
```diff
@@ -0,0 +1,117 @@
...
+
+    assert.strictEqual(
+      coCore,
+      coCoreAddress,
+      `CoCore address must be equal to ${coCoreAddress}.`,
+    );
+
+  });
+
+  it('should fail to set coCore address if its already set', async () => {
```
Comment (schemar) [inline test/gateway/safe_core/set_co_core_address.js]:
```suggestion
  it('should fail to set coCore address if it\'s already set', async () => {
```

**Comment by gulshanvasnani** [comment]
LGTM 👍
